docs/tekton_results.md

```diff
@@ -0,0 +1,52 @@
+# Tekton Results
+
+We have added support for installing and managing [Tekton Results](https://github.com/tektoncd/results) through Operator.
+To install Tekton Result on your cluster follow steps as given below:
+- Make sure Tekton Pipelines is installed on your cluster, using the Operator.
+- Generate a database root password.
+  A database root password must be generated and stored in a [Kubernetes Secret](https://kubernetes.io/docs/concepts/configuration/secret/)
+  before installing results. By default, Tekton Results expects this secret to have
+  the following properties:
+
+    - namespace: `tekton-pipelin  es`
+    - name: `tekton-results-mysql`
+    - contains the fields:
+        - `user=root`
+        - `password=<your password>`
+
+  If you are not using a particular password management strategy, the following
+  command will generate a random password for you:
+  Update namespace value in the command if Tekton Pipelines is installed in a different namespace..
+
+   ```sh
+   $ kubectl create secret generic tekton-results-mysql --namespace=tekton-pipelines --from-literal=user=root --from-literal=password=$(openssl rand -base64 20)
+   ```
+- Generate cert/key pair. 
+  Note: Feel free to use any cert management software to do this!
+
+  Tekton Results expects the cert/key pair to be stored in a [TLS Kubernetes Secret](https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets).
+  Update the namespace value in below export command if Tekton Pipelines is installed in a different namespace.
+   ```sh
+   # Generate new self-signed cert.
+   $ export NAMESPACE="tekton-pipelines"
+   $ openssl req -x509 \
+   -newkey rsa:4096 \
+   -keyout key.pem \
+   -out cert.pem \
+   -days 365 \
+   -nodes \
+   -subj "/CN=tekton-results-api-service.${NAMESPACE}.svc.cluster.local" \
+   -addext "subjectAltName = DNS:tekton-results-api-service.${NAMESPACE}.svc.cluster.local"
+   # Create new TLS Secret from cert.
+   $ kubectl create secret tls -n ${NAMESPACE} tekton-results-tls \
+   --cert=cert.pem \
+   --key=key.pem
+   ```
```

Comment:
I'm not super familiar with operator best practices, but would it be possible to automate these steps for the user if we find that they are not present at reconcile time?

For the TLS cert, are there existing examples for how we can hook into cert-manager or something similar?

Reply:
yeah that's the plan to automate this also, To start with we kept the creation of secrets a manual steps. so, user will create secrets and then TektonResult CR.
cc @nikhil-thomas

Reply:
> would it be possible to automate these steps for the user if we find that they are not present at reconcile time?
that is definitely the ideal ux the operator should deliver.
we will definitely handle in in upcoming iterations. 🧑‍💻 👍 

i have created an issue to track this: Automate creation prerequisites for TektonResult install #331